packages/contracts-bedrock/src/universal/ReinitializableBase.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/// @title ReinitializableBase
+/// @notice A base contract for reinitializable contracts that exposes a version number.
+abstract contract ReinitializableBase {
+    /// @notice Thrown when the initialization version is zero.
+    error ReinitializableBase_ZeroInitVersion();
+
+    /// @notice Current initialization version.
+    uint8 internal immutable INIT_VERSION;
```

Comment:
We should have a semgrep rule disallowing immutables and require a comment justifying why this is ok. We don't have the semgrep rule, but we should add the comment justification anyway

Reply:
To avoid issues with MCP?

Reply:
Yea exactly

Reply:
Ok I'll make an issue for this